backend/src/layer/data/flickr_api/json_parser.rs

```diff
@@ -40,32 +42,33 @@ impl JsonParser {
                 .map(|s| s.source.clone());
         }
 
-        match url {
-            None => return Err(ImageHosterError::ImageIsToSmall),
-            Some(url) => Ok(ImageMetaData {
+        url.map_or(Err(ImageHosterError::ImageIsToSmall), |url| {
+            Ok(ImageMetaData {
                 id: String::from(photo_id),
                 image_url: url,
-            }),
-        }
+            })
+        })
     }
 
     /// Obtains and validates the license by the information from the [`JsonRootLicense`] struct.
     /// # Return
     /// A boolean if the image has an valid license or not.
-    /// If the image has no license or no license history, the image isn't restricted by any license and true 'll be returned.
-    pub fn check_license(root: JsonRootLicense) -> bool {
+    /// If the image has no license or no license history, the image isn't restricted by any license and true will be returned.
+    #[must_use]
+    pub fn check_license(root: &JsonRootLicense) -> bool {
         let license = root
             .license_history
             .iter()
             .max_by_key(|l| l.date_change)
-            .map(|entry| entry.new_license)
+            .map(|entry| entry.new_license.clone())
             .unwrap_or_default();
         VALID_LICENSES.contains(&&*license)
```

Comment:
why doo you need to clone and then dereference and take two references?

Reply:
that map needs clone() as entry is just a reference (?) Idk. but that &&* is needed as contains needs &str and license is an String